lib/matplotlib/widgets.py

```diff
@@ -1085,7 +1087,7 @@ def __init__(self, ax, labels, actives=None, *, useblit=True,
 
     def _clear(self, event):
         """Internal event handler to clear the buttons."""
-        if self.ignore(event) or self.canvas.is_saving():
+        if self.ignore(event) or self.canvas is None or self.canvas.is_saving():
```

Comment:
Should the canvas is not None check rather go to a new `AxesWidget.ignore`?  (`def AxesWidget.ignore(self, event): return not self.active or self.canvas is None`)
Then this change and the one immediately below could go away.

Reply:
I agree. This belongs logically in an `AxesWidget.ignore()`.